MapboxSearch/sdk/src/main/java/com/mapbox/search/engine/OneStepRequestCallbackWrapper.kt

```diff
@@ -46,7 +47,7 @@ internal class OneStepRequestCallbackWrapper(
                     if (error != null) {
                         reportRelease(error)
                     } else {
-                        error = Exception("Unknown error. Response: $response")
+                        error = SearchRequestException(message = response.message, code = response.httpCode)
```

Comment:
With this code if we receive 400 or other HTTP error code in `HttpClientImpl` class, we don't report the exception to Telemetry SDK. IMO it's still a valuable use case to report errors so we could react to backend issues in a faster manner. Could we still report all failed HTTP requests?

Reply:
Good catch, thank you! I've returned old logic with reporting.

> if we receive 400 

Actually, it's a good topic to discuss, I'm not sure if we should report client errors (4xx). I'll cut a follow-up ticket for discussion.

Reply:
> Actually, it's a good topic to discuss, I'm not sure if we should report client errors (4xx)

AFAICS we send only `Exception.message` and we don't have any mention of HTTP code anywere in a report (unless this code is specified in the error message itself). So we should be fine!